fix(mock): prepend strict mock type declarations in tags-split MSW/faker output

## Problem

Orval 8.16.0 generates mock factory signatures that reference strict mock helper types (`MockWithNullableOverrides`, `{Schema}Mock`, etc.) but does not emit the corresponding `export type` declarations when all of the following are combined:

- `output.mode: 'tags-split'`
- `override.mock.required: true` and `override.mock.nonNullable: true` (strict mock mode, #3529)
- `mock.generators: [{ type: 'msw' }, { type: 'faker', arrayItems: true }]`
- `mock.indexMockFiles: true`

This produces TypeScript errors such as:

```
error TS2304: Cannot find name 'MockWithNullableOverrides'.
error TS2304: Cannot find name 'PetMock'.
error TS2552: Cannot find name 'getPetMock'. Did you mean 'getGetPetsResponseMock'?
```

## Why existing tests do not catch it

Strict-mock fixtures (`issue-3525`, `issue-3525-multi`, etc.) use `client: 'fetch'` and a single-file / non-tags-split layout. They do not cover tags-split per-tag `.msw.ts` + `.faker.ts` with `arrayItems: true`.

In that layout, `strictMockSchemaTypeNames` was dropped when building tag targets, so `finalizeMockImplementation` / `dedupeStrictMockTypeDeclarations` never prepended strict types. Faker files also referenced schema factories emitted only in the MSW file because array-item factory dedup was scoped per tag but not per mock generator file.

## Expected behavior

Each generated mock file that references strict mock types should include (once per file):

- `export type KeysWithNull<O> = { ... };`
- `export type MockWithNullableOverrides<...> = ...;`
- `export type PetMock = { [K in keyof Required<Pet>]: NonNullable<Required<Pet>[K]>; };`

Faker files must define or import schema factories they call (e.g. `getPetMock()`).

## Related PRs

- #3529 — strict mock return types when `required` + `nonNullable` are both true
- #3543 — emit strict types once via `strictMockSchemaTypeNames` / `finalizeMockImplementation`

--- a/packages/mock/src/mock-types.test.ts
+++ b/packages/mock/src/mock-types.test.ts
@@ -288,6 +288,19 @@ describe('mock-types', () => {
         collectStrictMockSchemaTypeNamesFromImplementation(implementation),
       ).toEqual(['WidgetMock']);
     });
+
+    it('keeps both Widget and WidgetMock when both schemas are strict-mocked', () => {
+      const implementation = [
+        'export const getWidgetMock = <O extends Partial<Widget> = {}>(overrideResponse?: O): MockWithNullableOverrides<Widget, O, WidgetMock> => ({}) as MockWithNullableOverrides<Widget, O, WidgetMock>;',
+        'export const getWidgetTypeMock = <O extends Partial<WidgetMock> = {}>(overrideResponse?: O): MockWithNullableOverrides<WidgetMock, O, WidgetMockMock> => ({}) as MockWithNullableOverrides<WidgetMock, O, WidgetMockMock>;',
+      ].join('\n');
+
+      expect(
+        collectStrictMockSchemaTypeNamesFromImplementation(
+          implementation,
+        ).toSorted(),
+      ).toEqual(['Widget', 'WidgetMock']);
+    });
   });
 
   describe('buildStrictMockTypeFileHeader', () => {

--- a/packages/mock/src/mock-types.ts
+++ b/packages/mock/src/mock-types.ts
@@ -222,11 +222,25 @@ export function applyStrictMockReturnType(
 
 const STRICT_MOCK_SCHEMA_TYPE_FROM_OVERRIDES =
   /MockWithNullableOverrides<([A-Z]\w*),/g;
-const STRICT_MOCK_SCHEMA_TYPE_FROM_MOCK_ALIAS = /\b([A-Z]\w*)Mock\b/g;
+const STRICT_MOCK_SCHEMA_TYPE_FROM_OVERRIDE_ALIAS =
+  /MockWithNullableOverrides<[^,]+,\s*[^,]+,\s*([A-Z]\w*Mock)>/g;
+const STRICT_MOCK_SCHEMA_TYPE_FROM_MOCK_ALIAS_RETURN =
+  /\): ([A-Z]\w*Mock)(?:\[\]|;)/g;
+
+/** Inverse of {@link getStrictMockTypeName}: `PetMock` → `Pet`, `WidgetMockMock` → `WidgetMock`. */
+function getSchemaTypeNameFromStrictMockAlias(alias: string): string {
+  return alias.endsWith('Mock') ? alias.slice(0, -4) : alias;
+}
 
 /**
  * Collect schema type names referenced by strict mock factories in generated
  * implementation text (nested split factories, array item helpers, etc.).
+ *
+ * This reverse-parses emitted factory syntax and is therefore coupled to the
+ * current `formatMockFactoryDeclaration` / `getMockFactorySignatureParts`
+ * shape. The structurally robust alternative is to record each nested item's
+ * schema name where split factories are generated (array-item / faker getters,
+ * where the `$ref` name is known) and thread it into `strictMockSchemaTypeNames`.
  */
 export function collectStrictMockSchemaTypeNamesFromImplementation(
   implementation: string,
@@ -239,16 +253,16 @@ export function collectStrictMockSchemaTypeNamesFromImplementation(
     names.add(match[1]);
   }
 
-  for (const match of implementation.matchAll(
-    STRICT_MOCK_SCHEMA_TYPE_FROM_MOCK_ALIAS,
-  )) {
-    names.add(match[1]);
+  for (const pattern of [
+    STRICT_MOCK_SCHEMA_TYPE_FROM_OVERRIDE_ALIAS,
+    STRICT_MOCK_SCHEMA_TYPE_FROM_MOCK_ALIAS_RETURN,
+  ]) {
+    for (const match of implementation.matchAll(pattern)) {
+      names.add(getSchemaTypeNameFromStrictMockAlias(match[1]));
+    }
   }
 
-  // `\b([A-Z]\w*)Mock\b` also matches `Widget` inside `WidgetMock` when the
-  // schema itself is named `WidgetMock` (`WidgetMockMock` strict alias). Drop
-  // prefix captures when a longer `{name}Mock` schema name was also collected.
-  return [...names].filter((name) => !names.has(`${name}Mock`));
+  return [...names];
 }
 
 export function mergeStrictMockSchemaTypeNames(

--- a/packages/mock/src/msw/index.ts
+++ b/packages/mock/src/msw/index.ts
@@ -430,6 +430,8 @@ export const ${handlerName} = (overrideResponse?: ${mockReturnType} | ((${infoPa
     strictMockSchemaTypeNames: strictMock
       ? mergeStrictMockSchemaTypeNames(
           schemaTypeNames,
+          // Nested split factories: see collectStrictMockSchemaTypeNamesFromImplementation
+          // re regex-coupling note — prefer threading names from getters long-term.
           collectStrictMockSchemaTypeNamesFromImplementation(
             mockImplementation,
           ),

